Add integration tests for SPIFFE SVIDs (#11)
rustls/webpki: tests/spiffe/README.txt

> An SVID is a structured x509 certificate that follows the SPIFFE x509-SVID

Reviewer: One last nit: I think we want this file to be `README.md` instead of `README.txt`.

format available at:

	https://github.com/spiffe/spiffe/blob/main/standards/X509-SVID.md

For these tests, an SVID was created following the example from: 

	https://spiffe.io/docs/latest/try/getting-started-linux-macos-x/

> ```

Reviewer: ```suggestion
```bash
```

git clone https://github.com/spiffe/spire
cd spire
go build ./cmd/spire-server
go build ./cmd/spire-agent
./spire-server run -config conf/server/server.conf&
./spire-server healthcheck
./spire-server token generate -spiffeID spiffe://example.org/myagent
./spire-agent run -config conf/agent/agent.conf -joinToken <token_string> &
./spire-agent healthcheck
./spire-server entry create -parentID spiffe://example.org/myagent \
    -spiffeID spiffe://example.org/myservice -selector unix:uid:$(id -u)
mkdir /tmp/spiffe-svid
./spire-agent api fetch x509 -write /tmp/spiffe-svid
```

The above commands generate three files
* `bundle.0.pem` contains the set of root CAs trusted by your current trust domain
* `svid.0.pem` contains the SVID (x509 certificate) and its associated chain of
  trust. In this case, the intermediate agent's own identity.
* `svid.0.key` contains the private key of the the SVID for `myservice`

You can inspect the above using

> ```
> openssl x509 -in bundle.0.pem -text
> openssl x509 -in svid.0.pem -text # only shows the first certificate, which is the SVID
> ```

Reviewer: ```suggestion
```bash
openssl x509 -in /tmp/spiffe-svid/bundle.0.pem -text
openssl x509 -in /tmp/spiffe-svid/svid.0.pem -text # only shows the first certificate, which is the SVID
```
```

> For webpki tests, I split the svid.0.pem into two separate files:

Reviewer: nit: avoiding first person.
```suggestion
For webpki tests, the `svid.0.pem` file is split into two separate files:
```

 * inter.0.pem
 * service.0.pem

> I did the above manually since it's a one-off.

Reviewer: nit: avoiding first person.
```suggestion
This is done manually since it's a one-off.
```

```

> openssl x509 -in bundle.0.pem -out bundle.0.der -outform DER 
> openssl x509 -in inter.0.pem -out inter.0.der -outform DER 
> openssl x509 -in service.0.pem -out service.0.der -outform DER 

Reviewer: ```suggestion
openssl x509 -in /tmp/spiffe-svid/bundle.0.pem -out /tmp/spiffe-svid/bundle.0.der -outform DER 
openssl x509 -in /tmp/spiffe-svid/inter.0.pem -out /tmp/spiffe-svid/inter.0.der -outform DER 
openssl x509 -in /tmp/spiffe-svid/service.0.pem -out /tmp/spiffe-svid/service.0.der -outform DER 
```

```

> I then cleaned up the directory a bit by deleting the intermediate inter.0.pem
> and service.0.pem since these are preserved in the included svid.0.pem.

Reviewer: editorial nits:
```suggestion
The final test content is created by deleting the intermediate `inter.0.pem`
and `service.0.pem` files since these are preserved in the included `svid.0.pem`.
```
